pkg/azuredisk/controllerserver_v2.go

```diff
@@ -897,57 +962,15 @@ func (d *DriverV2) ListSnapshots(ctx context.Context, req *csi.ListSnapshotsRequ
 }
 
 func (d *DriverV2) getSnapshotByID(ctx context.Context, resourceGroup, snapshotID, sourceVolumeID string) (*csi.Snapshot, error) {
-	var err error
-	snapshotName := snapshotID
-	if isARMResourceID(snapshotID) {
-		snapshotName, resourceGroup, err = d.getSnapshotInfo(snapshotID)
-		if err != nil {
-			return nil, err
-		}
-	}
-
-	snapshot, rerr := d.cloud.SnapshotsClient.Get(ctx, resourceGroup, snapshotName)
-	if rerr != nil {
-		return nil, status.Error(codes.Internal, fmt.Sprintf("get snapshot %s from rg(%s) error: %v", snapshotName, resourceGroup, rerr.Error()))
-	}
-
-	return generateCSISnapshot(sourceVolumeID, &snapshot)
+	return nil, nil
 }
 
 // GetSourceDiskSize recursively searches for the sourceDisk and returns: sourceDisk disk size, error
 func (d *DriverV2) GetSourceDiskSize(ctx context.Context, resourceGroup, diskName string, curDepth, maxDepth int) (*int32, error) {
-	if curDepth > maxDepth {
-		return nil, status.Error(codes.Internal, fmt.Sprintf("current depth (%d) surpassed the max depth (%d) while searching for the source disk size", curDepth, maxDepth))
-	}
-	result, rerr := d.cloud.DisksClient.Get(ctx, resourceGroup, diskName)
-	if rerr != nil {
-		return nil, rerr.Error()
-	}
-	if result.DiskProperties == nil {
-		return nil, status.Error(codes.Internal, fmt.Sprintf("DiskProperty not found for disk (%s) in resource group (%s)", diskName, resourceGroup))
-	}
-
-	if result.DiskProperties.CreationData != nil && (*result.DiskProperties.CreationData).CreateOption == "Copy" {
-		klog.V(2).Infof("Clone source disk has a parent source")
-		sourceResourceID := *result.DiskProperties.CreationData.SourceResourceID
-		parentResourceGroup, _ := GetResourceGroupFromURI(sourceResourceID)
-		parentDiskName := path.Base(sourceResourceID)
-		return d.GetSourceDiskSize(ctx, parentResourceGroup, parentDiskName, curDepth+1, maxDepth)
-	}
-
-	if (*result.DiskProperties).DiskSizeGB == nil {
-		return nil, status.Error(codes.Internal, fmt.Sprintf("DiskSizeGB for disk (%s) in resourcegroup (%s) is nil", diskName, resourceGroup))
-	}
-	return (*result.DiskProperties).DiskSizeGB, nil
+	return 0, nil
 }
 
 // The format of snapshot id is /subscriptions/xxx/resourceGroups/xxx/providers/Microsoft.Compute/snapshots/snapshot-xxx-xxx.
 func (d *DriverV2) getSnapshotInfo(snapshotID string) (snapshotName, resourceGroup string, err error) {
-	if snapshotName, err = getSnapshotName(snapshotID); err != nil {
-		return "", "", err
-	}
-	if resourceGroup, err = GetResourceGroupFromURI(snapshotID); err != nil {
-		return "", "", err
-	}
-	return snapshotName, resourceGroup, err
+	return "", "", nil
```

Comment:
Why is this now a no-op?

Reply:
These particular functions getSnapshotByID, GetSourceDiskSize, getSnapshotInfo has been defined with receivers Driver in V1 and V2. 
The use case these functions cover are strictly related to azure, hence I have moved them to azureutils pkg.

Ideally, I dont see the reason behind these functions being a part of the interface Driver. 
Because, we are leaving the V1 unchanged, I have now changed these functions to no-op to maintain the coherence of the interface.

Reply:
Ah, I see. Yes, this currently needs to be part of the driver interface because of the unit test that covers it. You should at least add a comment describing the reason for the no-op and maybe even log a warning trace. The unit test should also be disabled when testing v2 and a corresponding unit test added to the provisioner package.

Reply:
You could also go ahead an change the v1 implementation to use the refactored `azureutils` methods and remove these stub methods from the v2 implementation and common interface. This is a change to the v1 driver, but should be fairly small and not likely to cause many integration issues.